### PR TITLE
Use CreateTags instead of TagResources

### PR DESCRIPTION
**Pull request by nckturner**
*Issue #, if available:*
Fixes: https://github.com/aws/amazon-vpc-cni-k8s/issues/76

*Description of changes:*

* Switches to CreateTags instead of TagResources.
* Also introduce two new tags:
  - node.k8s.amazonaws.com/instance_id
  - cluster.k8s.amazonaws.com/name (only tagged if the env var CLUSTER_NAME is present).

*Testing:*
Created an EKS cluster with a node, which was created with the current CNI.  I  checked the tag on the ENI, it was the old format.  I then applied CNI yaml with a container I built and scaled in and out the autoscaling group to get new nodes.  The first time I did this with CLUSTER_NAME set:

tag key | tag value
-- | --
cluster.k8s.amazonaws.com/name | wonderful-outfit-1531613390
node.k8s.amazonaws.com/instance_id | i-03eb4524c4c96774a

Second time without: 

tag key | tag value
-- | --
node.k8s.amazonaws.com/instance_id | i-0bf1f4f8e688b70fd

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/awsutils/awsutils.go [hunk 1]
```diff
@@ -625,37 +621,40 @@ func (cache *EC2InstanceMetadataCache) createENI() (string, error) {
...
+			Key:   aws.String(eniClusterTagKey),
+			Value: aws.String(clusterName),
+		})
+	}
 
-	tagValue := cache.instanceID
-	tags[eniTagKey] = aws.String(tagValue)
-	log.Debugf("Trying to tag newly created eni: keys=%s, value=%s", eniTagKey, tagValue)
+	for _, tag := range tags {
+		log.Debugf("Trying to tag newly created eni: key=%s, value=%s", *tag.Key, *tag.Value)
```
Comment (vsiddharth) [inline pkg/awsutils/awsutils.go]: Can you use `aws.StringValue` here?

File: pkg/awsutils/awsutils.go [hunk 2]
```diff
@@ -625,37 +621,40 @@ func (cache *EC2InstanceMetadataCache) createENI() (string, error) {
...
 	start := time.Now()
-	_, err := tagSvc.TagResources(tagInput)
-	awsAPILatency.WithLabelValues("TagResources", fmt.Sprint(err != nil)).Observe(msSince(start))
+	_, err := cache.ec2SVC.CreateTags(input)
+	awsAPILatency.WithLabelValues("CreateTags", fmt.Sprint(err != nil)).Observe(msSince(start))
 	if err != nil {
-		awsAPIErrInc("TagResources", err)
-		log.Warnf("Fail to tag the newly created eni %s  %v", eniID, err)
+		awsAPIErrInc("CreateTags", err)
+		log.Warnf("Fail to tag the newly created eni %s:  %v", eniID, err)
```
Comment (vsiddharth) [inline pkg/awsutils/awsutils.go]: s/Fail/Failed

File: pkg/awsutils/awsutils.go [hunk 3]
```diff
@@ -625,37 +621,40 @@ func (cache *EC2InstanceMetadataCache) createENI() (string, error) {
...
-
-	eniARN := &arn.ARN{
-		Partition: "aws",
-		Service:   "ec2",
-		Region:    cache.region,
-		AccountID: cache.accountID,
-		Resource:  "network-interface/" + eniID}
-	arnString := eniARN.String()
-	arns = append(arns, aws.String(arnString))
+	tags := []*ec2.Tag{
```
Comment (vsiddharth) [inline pkg/awsutils/awsutils.go]: Can you add a comment with the sample tag keys and values for an ENI??

File: pkg/awsutils/awsutils.go [hunk 4]
```diff
@@ -625,37 +621,40 @@ func (cache *EC2InstanceMetadataCache) createENI() (string, error) {
...
+	_, err := cache.ec2SVC.CreateTags(input)
+	awsAPILatency.WithLabelValues("CreateTags", fmt.Sprint(err != nil)).Observe(msSince(start))
 	if err != nil {
-		awsAPIErrInc("TagResources", err)
-		log.Warnf("Fail to tag the newly created eni %s  %v", eniID, err)
+		awsAPIErrInc("CreateTags", err)
+		log.Warnf("Fail to tag the newly created eni %s:  %v", eniID, err)
 	} else {
-		log.Debugf("Tag the newly created eni with arn: %s", arnString)
+		log.Debugf("Tag the newly created eni: %s", eniID)
```
Comment (vsiddharth) [inline pkg/awsutils/awsutils.go]: Can you please rephrase this message to
`"Successfully tagged eni: %s", eniID`

File: pkg/awsutils/awsutils.go (hunk 1 above)
Comment (nckturner) [reply pkg/awsutils/awsutils.go]: Done.

File: pkg/awsutils/awsutils.go (hunk 2 above)
Comment (nckturner) [reply pkg/awsutils/awsutils.go]: Good catch, done.

File: pkg/awsutils/awsutils.go (hunk 3 above)
Comment (nckturner) [reply pkg/awsutils/awsutils.go]: Yeah.

File: pkg/awsutils/awsutils.go (hunk 4 above)
Comment (nckturner) [reply pkg/awsutils/awsutils.go]: Yup, makes more sense.

**Pull request merged**